msal/src/main/java/com/microsoft/identity/client/TokenParameters.java

```diff
@@ -202,6 +204,18 @@ public B callback(AuthenticationCallback callback) {
             return self();
         }
 
+        public B withResource(final String resource) {
+            if (!StringUtil.isEmpty(resource)) {
+                mScopes = new ArrayList<String>() {{
+                    add(resource.toLowerCase().trim() + "/.default");
+                }};
+            } else {
+                mScopes = new ArrayList<>();
```

Comment:
When would this ever be valid? We should throw an `IllegalArgumentException`, rather than let developers pass null and get some _other_ cryptic error elsewhere